src/tests/test_connection.py

```diff
@@ -576,3 +577,216 @@ def test_feedback_when_switching_connection_without_alias(ip_empty, tmp_empty, c
+
+    yield mock.execute, conn_sqlalchemy
+
+
+@pytest.fixture
+def mock_dbapi_raw_execute(monkeypatch, conn_dbapi):
+    mock = Mock()
+    monkeypatch.setattr(conn_dbapi, "_connection", mock)
+    # mock the dialect to pretend we're using tsql
+    monkeypatch.setattr(conn_dbapi, "_get_sqlglot_dialect", lambda: "tsql")
+
+    yield mock.cursor().execute, conn_dbapi
+
+
+@pytest.mark.parametrize(
+    "fixture_name",
+    [
+        "mock_sqlalchemy_raw_execute",
+        "mock_dbapi_raw_execute",
+    ],
+)
+def test_raw_execute_doesnt_transpile_sql_query(fixture_name, request):
+    mock_execute, conn = request.getfixturevalue(fixture_name)
+
+    conn.raw_execute("CREATE TABLE foo (bar INT)")
+    conn.raw_execute("INSERT INTO foo VALUES (42), (43)")
+    conn.raw_execute("SELECT * FROM foo LIMIT 1")
+
+    calls = [
+        str(call[0][0])
+        for call in mock_execute.call_args_list
+        # if running on sqlalchemy 1.x, the commit call is done via .execute,
+        # ignore them
+        if str(call[0][0]) != "commit"
+    ]
+
+    expected_number_of_calls = 3
+    expected_calls = [
+        "CREATE TABLE foo (bar INT)",
+        "INSERT INTO foo VALUES (42), (43)",
+        "SELECT * FROM foo LIMIT 1",
+    ]
+
+    assert len(calls) == expected_number_of_calls
+    assert calls == expected_calls
+
+
+@pytest.fixture
+def mock_sqlalchemy_execute(monkeypatch):
+    conn = SQLAlchemyConnection(engine=create_engine("duckdb://"))
+
+    mock = Mock()
+    monkeypatch.setattr(conn._connection, "execute", mock)
+    # mock the dialect to pretend we're using tsql
+    monkeypatch.setattr(conn, "_get_sqlglot_dialect", lambda: "tsql")
+
+    yield mock, conn
+
+
+@pytest.fixture
+def mock_dbapi_execute(monkeypatch):
+    conn = DBAPIConnection(duckdb.connect())
+
+    mock = Mock()
+    monkeypatch.setattr(conn, "_connection", mock)
+    # mock the dialect to pretend we're using tsql
+    monkeypatch.setattr(conn, "_get_sqlglot_dialect", lambda: "tsql")
+
+    yield mock.cursor().execute, conn
+
+
+@pytest.mark.parametrize(
+    "fixture_name",
+    [
+        "mock_sqlalchemy_execute",
+        "mock_dbapi_execute",
+    ],
+    ids=[
+        "sqlalchemy",
+        "dbapi",
+    ],
+)
+def test_execute_transpiles_sql_query(fixture_name, request):
+    mock_execute, conn = request.getfixturevalue(fixture_name)
+
+    conn.execute("CREATE TABLE foo (bar INT)")
+    conn.execute("INSERT INTO foo VALUES (42), (43)")
+    conn.execute("SELECT * FROM foo LIMIT 1")
+
+    calls = [
+        str(call[0][0])
+        for call in mock_execute.call_args_list
+        # if running on sqlalchemy 1.x, the commit call is done via .execute,
+        # ignore them
+        if str(call[0][0]) != "commit"
+    ]
+
+    expected_number_of_calls = 3
+    expected_calls = [
+        "CREATE TABLE foo (bar INTEGER)",
+        "INSERT INTO foo VALUES (42), (43)",
+        # since we're transpiling, we should see TSQL code
+        "SELECT TOP 1 * FROM foo",
```

Comment:
is there a way to verify the transpiled query?

Reply:
as @yafimvo suggests, the way o verify the transpiled query is to run it, which we do in the integration tests. so there's no need to verify this one, as it's just a made-up example. the ones we need to verify are in the integration tests